src/main.rs

```diff
@@ -0,0 +1,255 @@
+    )]
+    sketches_reports_source: PathBuf,
+
+    /// The GitHub access token used to post comments on the PR thread
+    #[arg(short, long, env = "GITHUB_TOKEN")]
+    token: Option<String>,
+}
+
+struct Logger;
+
+impl Logger {
+    fn level_color(level: &Level) -> String {
+        let name = format!("{:>5}", level.as_str().to_uppercase());
+        match level {
+            Level::Error => name.red().bold().to_string(),
+            Level::Warn => name.yellow().bold().to_string(),
+            Level::Info => name.green().bold().to_string(),
+            Level::Debug => name.blue().bold().to_string(),
+            Level::Trace => name.magenta().bold().to_string(),
+        }
+    }
+}
+
+impl log::Log for Logger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= log::max_level()
+    }
+
+    fn log(&self, record: &Record) {
+        let mut stdout = stdout().lock();
+        if record.target() == "CI_LOG_GROUPING" {
+            // this log is meant to manipulate a CI workflow's log grouping
+            writeln!(stdout, "{}", record.args()).expect("Failed to write log command to stdout");
+            stdout
+                .flush()
+                .expect("Failed to flush log command to stdout");
+        } else if self.enabled(record.metadata()) {
+            let module = record.module_path();
+            if module.is_none_or(|v| {
+                v.starts_with("arduino_report_size_deltas") || v.starts_with("report_size_deltas")
+            }) {
+                writeln!(
+                    stdout,
+                    "[{}]: {}",
+                    Self::level_color(&record.level()),
+                    record.args()
+                )
+                .expect("Failed to write log message to stdout");
+            } else {
+                writeln!(
+                    stdout,
+                    "[{}]{{{}:{}}}: {}",
+                    Self::level_color(&record.level()),
+                    module.unwrap(), // safe to unwrap here because the None case is caught above
+                    record.line().unwrap_or_default(),
+                    record.args()
+                )
+                .expect("Failed to write detailed log message to stdout");
+            }
+            stdout
+                .flush()
+                .expect("Failed to flush log message to stdout");
+        }
+    }
+
+    fn flush(&self) {}
+}
+
+static LOGGER: Logger = Logger;
+
+pub fn logger_init() {
+    let _ = log::set_logger(&LOGGER);
+}
+
+async fn run(args: &[String]) -> Result<()> {
+    let args = Args::parse_from(args);
+    logger_init();
+    let client =
+        GithubApiClient::new().with_context(|| "Failed to instantiate GitHub REST API client")?;
+    log::set_max_level(if client.debug_enabled {
+        LevelFilter::Debug
+    } else {
+        LevelFilter::Info
+    });
+
+    GithubApiClient::start_log_group("Generating comment from JSON files");
```

Comment:
Should check if event is a PR and exit early if not:
```suggestion
    if !client.is_pr_event() {
        log::info!("Comment only gets posted to a PR");
        return Ok(());
    }

    GithubApiClient::start_log_group("Generating comment from JSON files");
```

Reply:
done with tests and using step summary instead; https://github.com/2bndy5/arduino-report-size-deltas/pull/1/commits/97279567c690e926983b12e072c59ee803e8aec7